src/coreComponents/python/modules/geosx_mesh_doctor/checks/reorient_mesh.py

```diff
@@ -0,0 +1,171 @@
+    Given a polyhedra, given that we were able to paint the faces in two colors,
+    we now need to select which faces/color to flip such that the volume of the element is positive.
+    :param mesh_points: The mesh points, needed to compute the volume.
+    :param colors: Maps the nodes of each connected component (defined as a frozenset) to its color.
+    :param face_stream: the polyhedron.
+    :return: The face stream that leads to a positive volume.
+    """
+    # Flipping either color 0 or 1.
+    color_to_nodes: Dict[int, List[int]] = {0: [], 1: []}
+    for connected_components_indices, color in colors.items():
+        color_to_nodes[color] += connected_components_indices
+    # This implementation works even if there is one unique color.
+    # Admittedly, there will be one face stream that won't be flipped.
+    fs = face_stream.flip_faces(color_to_nodes[0]), face_stream.flip_faces(color_to_nodes[1])
+    volumes = __compute_volume(mesh_points, fs[0]), __compute_volume(mesh_points, fs[1])
+    # We keep the flipped element for which the volume is largest
+    # (i.e. positive, since they should be the opposite of each other).
+    return fs[numpy.argmax(volumes)]
+
+
+def __reorient_element(mesh_points: vtkPoints, face_stream_ids: vtkIdList) -> vtkIdList:
+    """
+    Considers a vtk face stream and flips the appropriate faces to get an element with normals directed outwards.
+    :param mesh_points: The mesh points, needed to compute the volume.
+    :param face_stream_ids: The raw vtk face stream, not converted into a more practical python class.
+    :return: The raw vtk face stream with faces properly flipped.
+    """
+    face_stream = FaceStream.build_from_vtk_id_list(face_stream_ids)
+    face_graph = build_cell_graph(face_stream, add_compatibility=True)
+    # Removing the non-compatible connections to build the non-connected components.
+    g = networkx.Graph()
+    g.add_nodes_from(face_graph.nodes)
+    g.add_edges_from(filter(lambda uvd: uvd[2]["compatible"] == "+", face_graph.edges(data=True)))
+    connected_components = tuple(networkx.connected_components(g))
+    # Squashing all the connected nodes that need to receive the normal direction flip (or not) together.
+    quotient_graph = networkx.algorithms.quotient_graph(face_graph, connected_components)
+    # Coloring the new graph lets us know how which cluster of faces need to eventually receive the same flip.
+    # W.r.t. the nature of our problem (a normal can be directed inwards or outwards),
+    # two colors should be enough to color the face graph.
+    # `colors` maps the nodes of each connected component to its color.
+    colors: Dict[FrozenSet[int], int] = networkx.algorithms.greedy_color(quotient_graph)
+    assert len(colors) in (1, 2)
+    # We now compute the face stream which generates outwards normal vectors.
+    flipped_face_stream = __select_flip(mesh_points, colors, face_stream)
+    return to_vtk_id_list(flipped_face_stream.dump())
```

Comment:
Just to understand, how is the cell volume used in these lines of code?

Reply:
The normals can be whether all directed outwards or all inwards.
For the outwards case, the volume will be positive. And volume will be negative (and even the opposite) for inwards normals.

`__select_flip` computes the volumes for both cases and takes the configuration which generates the positive volume: normals will be facing outwards.